supabase/migrations/20240412000067_prompts-experiments.sql

```diff
@@ -0,0 +1,125 @@
+CREATE TABLE "public"."prompt_v2" (
+    id UUID PRIMARY KEY default gen_random_uuid(),
+    user_defined_id TEXT NOT NULL,
+    "description" TEXT,
+    pretty_name TEXT,
+    organization UUID NOT NULL,
+    CONSTRAINT unique_user_defined_id UNIQUE (user_defined_id, organization),
+    CONSTRAINT fk_organization FOREIGN KEY (organization) REFERENCES organization(id),
+    CONSTRAINT check_user_defined_id_length CHECK (LENGTH(user_defined_id) <= 32),
+    CONSTRAINT check_description_length CHECK (LENGTH(description) <= 512),
+    CONSTRAINT check_pretty_name_length CHECK (LENGTH(pretty_name) <= 128)
+);
+
+alter table "public"."prompt_v2" enable row level security;
+
+CREATE TABLE "public"."prompts_versions" (
+    id UUID PRIMARY KEY default gen_random_uuid(),
+    major_version INT NOT NULL,
+    minor_version INT NOT NULL,
+    soft_delete BOOLEAN DEFAULT FALSE,
+    helicone_template JSONB,
+    prompt_v2 UUID NOT NULL,
+    model TEXT,
+    organization UUID NOT NULL,
+    CONSTRAINT fk_prompt FOREIGN KEY (prompt_v2) REFERENCES prompt_v2(id),
+    CONSTRAINT fk_organization FOREIGN KEY (organization) REFERENCES organization(id),
+    CONSTRAINT check_major_version CHECK (major_version >= 0),
+    CONSTRAINT check_minor_version CHECK (minor_version >= 0)
+);
+CREATE INDEX idx_prompts_versions_major ON prompts_versions(organization, major_version DESC);
+CREATE INDEX idx_prompts_versions_minor ON prompts_versions(organization, major_version DESC, minor_version DESC);
+CREATE UNIQUE INDEX idx_prompts_versions_model ON prompts_versions(organization, prompt_v2, major_version DESC, minor_version DESC);
```

Comment:
unique index includes DESC for the versions?

Reply:
I guess it doesn't really matter, I will kill it.